game/effect.go

```diff
@@ -1,8 +1,49 @@
+/*
+	An Effect can be created by a spell, a triggered ability, or an
+	activated ability.
+
+	An Effect can be the property of a Card. For example, when the
+	Card is a Giant Growth, its effect property would be
+	&Effect{power:3, toughness:3}.
+
+	An Effect can also be the Morbid or Kicker property of a Card, to designate a
+	Effect that only happens under special circumstances.
+
+*/
+
 package game
 
 import ()
 
 type Effect struct {
-	Action *Action
-	Card   *Card
+
+	// from Card
+	CastingCost        *CastingCost // when an Effect is a kicker, it has a Cost
+	Hexproof           bool
+	Kicker             *Effect
```

Comment:
effects can have kickers that are other effects? that doesn't seem right

Reply:
Yeah, because when you make an Effect from a Card+Action, you do Effect.Kicker = Card.Kicker.

So the Effect might both have some Plus1Plus1Counters in itself, or in its child Kicker Effect.

Reply:
so the rule is when you resolve an Effect, you should also resolve any effect under Kicker? or do you re-check whether the kicker was paid? it seems like the checking of kicker should happen at casting time, not like every time you observe that a creature has a rancor on it. in magic it doesn't seem like permanents remember whether they were kicked.

Reply:
I've now set it up so Kicker only exists on an Effect if it's kicked.

So when you resolve an Effect, if it has a Kicker prop, you do that on top of the normal Effect.